lib/pbench/agent/tool_meister.py

```diff
@@ -1014,6 +1023,82 @@ def end_tools(self, data):
+        cmd="ssh ${ssh_opts} -n ${remote_host} pbench-remote-sysinfo-dump ${sysinfo_path} \"${sysinfo}\" ${remote_label}"
+        debug_log "[${script_name}] ${cmd} > ${sysinfo_path}/${remote_label}${remote_host}.tar 2> ${sysinfo_path}/${remote_label}${remote_host}.err"
+        ${cmd} > ${sysinfo_path}/${remote_label}${remote_host}.tar 2> ${sysinfo_path}/${remote_label}${remote_host}.err
+        if [[ ${?} -ne 0 ]]; then
+                error_log "Remote sysinfo-dump failed"
+                error_log "$(cat ${sysinfo_path}/${remote_label}${remote_host}.err)"
+        else
+                debug_log "[${script_name}]completed: deleting sysinfo-dump data on ${remote_host}"
+                debug_log "$(cat ${sysinfo_path}/${remote_label}${remote_host}.err)"
+                pushd "${sysinfo_path}" >/dev/null
+                tar xf ${remote_label}${remote_host}.tar
+                tar_status=${?}
+                popd >/dev/null
+                if [[ ${tar_status} -ne 0 ]]; then
+                        error_log "Remote tar ball was bad: ${sysinfo_path}/${remote_label}${remote_host}.tar"
+                fi
+        fi
+        rm -f ${sysinfo_path}/${remote_label}${remote_host}.tar
+        rm -f ${sysinfo_path}/${remote_label}${remote_host}.err
+}
+
+for dirent in $(/bin/ls -1 ${tool_group_dir}); do
+        if [[ "${dirent}" == "__trigger__" ]]; then
+                # Ignore trigger files
+                continue
+        elif [[ ! -d ${tool_group_dir}/${dirent} ]]; then
+                # Skip spurious files of ${tool_group_dir}
+                warn_log "[$script_name] \"${this_tool_file}\" is a file in \"${tool_group_dir}\"; that should not happen. Please consider deleting it."
+                continue
+        fi
+        hostdir="${dirent}"
+        label="$(cat ${tool_group_dir}/${hostdir}/__label__ 2> /dev/null)"
+        if [[ "${hostdir}" == "localhost" || "${hostdir}" == "${_pbench_hostname}" || "${hostdir}" == "${_pbench_full_hostname}" ]]; then
+                # We found local tools specified in the tools group directory,
+                # so we should collect sysinfo data locally.
+                pbench-sysinfo-dump "${sysinfo_path}" "${sysinfo}" "${label}" &
+        else
+                # We have a remote directory.
+                gather_remote_sysinfo_data ${hostdir} ${label} &
+        fi
+done
+wait
+"""
+        if self.state in ("running", "idle"):
+            # The "gather system information" action is only allowed when the
+            # Tool Meister first starts ("startup"), and when it is ready for
+            # shutting down ("shutdown").
+            msg = f"sysinfo action received in state '{self.state}'"
+            self._send_client_status(msg)
+            return 1
+
+        args = data["args"]
+        if not args:
+            msg = f"No sysinfo arguments given"
+            self._send_client_status(msg)
+            return 1
+
+        # FIXME - run the sysinfo collection
```

Comment:
Ah; or did you just leave the entire bash script because you haven't actually translated it yet?

Reply:
So I have implemented the collection ... but have not sent it remotely yet.